source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -237,7 +308,11 @@ FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool end_st
 
   FilterHeadersStatus status = FilterHeadersStatus::Continue;
   if (decoding_state_.sendHeaders()) {
-    status = onHeaders(decoding_state_, headers, end_stream);
+    auto activation_ptr = Filters::Common::Expr::createActivation(decoding_state_.streamInfo(),
+                                                                  &headers, nullptr, nullptr);
```

Comment:
for my own education, why do we need to use CEL parser to forward the attributes? Arent these available as part of streamInfo readily? I think @tyxia 's concern was the speed and overhead when we drag the CEL parser into the picture

Reply:
Attributes, as documented [here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes.html), are only implemented in CEL. I don't know why this very convenient mode of accessing this info was coupled to CEL in the first place, but because it is, CEL is used to access them here in lieu of re-implementing attributes in a more common/accessible location

Reply:
You may want to make conditional on presence of attributes in the configuration. Otherwise the CEL context is instantiated even if no attributes were needed.

Reply:
@yanavlasov I have made the change so the CEL context is only instantiated when needed